dashboard/app/controllers/api_controller.rb

```diff
@@ -153,6 +153,100 @@ def section_text_responses
     render json: data
   end
 
+  def section_assessments
+    load_section
+    load_script
+
+    level_group_script_levels = @script.script_levels.includes(:level).where("levels.type" => LevelGroup)
+
+    data = @section.students.map do |student|
+      student_hash = {id: student.id, name: student.name}
+
+      level_group_script_levels.map do |script_level|
+        next unless script_level.long_assessment?
+
+        last_attempt = student.last_attempt(script_level.level)
+        response = last_attempt.try(:level_source).try(:data)
+
+        next unless response
+
+        response_parsed = JSON.parse(response)
+
+        user_level = student.user_level_for(script_level)
+
+        # Summarize some key data.
+        multi_count = 0
+        multi_count_correct = 0
+
+        # And construct a listing of all the individual levels and their results.
+        level_results = []
+
+        next unless properties = script_level.level.properties
```

Comment:
````
next unless properties = script_level.level.properties
 +
 +        # Go through each page of the long assessment's LevelGroup.
 +        properties["pages"].each do |page|
 +          # Go through each level of the page.
 +          page["levels"].each do |page_level_name|
 +            level = Level.find_by_name(page_level_name)
````
these five lines could probably be extracted into a LevelGroup#levels method (I would be surprised if that is not used elsewhere)